Example/Messaging/Tests/FIRMessagingRmqManagerTest.m

```diff
@@ -302,6 +320,31 @@ - (void)testDeleteSyncMessage {
   XCTAssertNil([self.rmqManager querySyncMessageWithRmqID:rmqID]);
 }
 
+- (void)testInitWhenDatabaseIsBrokenThenDatabaseIsDeleted {
+  NSString *databaseName = @"invalid-database-file";
+  NSString *databasePath = [self createBrokenDatabaseWithName:databaseName];
+  XCTAssert([[NSFileManager defaultManager] fileExistsAtPath:databasePath]);
+
+  // Expect for at least one assertion.
```

Comment:
"Expect at least one"? or "Expect one assertion"?

Reply:
If there can only be one assertion in the `initWithDatabaseName`, it's likely worth having it only allow assertion. Otherwise we could be catching other assertions happening.

Reply:
Makes sense. I'll take a closer look how many assertions are supposed to fail and update the test

Reply:
Currently there are two DB query initiated on the object init. It looks like if the DB file is damaged for some reason, the class will not work as expected until FCM stack is initialized (basically until the host app restart I guess). I am not sure it is the best strategy. I assume we can improve it, but I hesitate doing it without a discussion with @chliangGoogle.

Taking into account that this behaviour was implemented a while ago and the deadlock was introduced just recently, I would proceed with the small fix for the deadlock (and keep the tests tolerant to other potential assertions) and consider a better recovery strategy in the future. WDYT?

Reply:
I think we added dispatch_async in removeDatabase recently for unit tests which is what i'm worried why deadlock happens.